### PR TITLE
feat: add qc step

### PR DESCRIPTION
**Pull request by addramir**
## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

**Comment by addramir** [comment]
Should work now

File: src/gentropy/sumstat_qc_step.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        """
+        gwas = SummaryStatistics.from_parquet(session, path=gwas_path)
+
+        QC = SummaryStatisticsQC.get_quality_control_metrics(
+            gwas=gwas, limit=100_000_000, min_count=100, n_total=100000
+        )
+        QC.write.mode(session.write_mode).parquet(
+            output_path + "/qc_results_" + studyid
+        )
```
Comment (DSuveges) [inline src/gentropy/sumstat_qc_step.py:36]: Very marginal comment, has no impact on the logic, but I would just write:

```python
(
    SummaryStatistics.from_parquet(session, path=gwas_path)
    .get_quality_control_metrics(
        gwas=gwas, limit=100_000_000, min_count=100, n_total=100000
    )
    .write.mode(session.write_mode).parquet(
        output_path + "/qc_results_" + studyid
    )
)
```

File: src/gentropy/sumstat_qc_step.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    """Step to run GWAS QC."""
+
+    def __init__(
+        self,
+        session: Session,
+        gwas_path: str,
+        output_path: str,
+        studyid: str,
+    ) -> None:
+        """Run fine-mapping on a studyLocusId from a collected studyLocus table.
```
Comment (DSuveges) [inline src/gentropy/sumstat_qc_step.py:20]: The first line needs to be changed. 

```suggestion
        """Step to calculate quality control metrics on the provided GWAS study.
```

File: src/gentropy/sumstat_qc_step.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+"""Step to run a finemapping using."""
```
Comment (DSuveges) [inline src/gentropy/sumstat_qc_step.py:1]: Needs to be changed:

```suggestion
"""Step to calculate quality control metrics on the provided GWAS study."""
```

**Review by DSuveges — CHANGES_REQUESTED** [review]
All seems to be OK, very minor changes in the documentation. Also there needs to be a markdown file added to `docs/python_api/steps/GWASQC.md`, so the documentation of this step shows up in `https://opentargets.github.io/gentropy/python_api/steps/_steps/`

File: src/gentropy/sumstat_qc_step.py (hunk 2 above)
Comment (addramir) [reply src/gentropy/sumstat_qc_step.py:20]: I was in a hurry for this PR, sorry for these left-overs from fine-mapping

Comment (DSuveges) [reply src/gentropy/sumstat_qc_step.py]: No problem at all. There are reviews for a reason.

File: src/gentropy/sumstat_qc_step.py (hunk 1 above)
Comment (addramir) [reply src/gentropy/sumstat_qc_step.py]: Done.

**Review by DSuveges — APPROVED** [review]

**Pull request merged**